src/GroupDetails.ts

```diff
@@ -0,0 +1,63 @@
+import { GroupHeading } from './Group';
+import type { Task } from './Task';
+
+export class IntermediateTaskGroupsStorage extends Map<string[], Task[]> {}
+
+export class IntermediateTaskGroups {
```

Comment:
Why intermediate? Looks like it shouldn't be exported?

Reply:
This is one of the areas where I need to add comments to clarify the implementation...
Am hoping someone will then be able to come up with a clearer implementation.

Reply:
I hope that the comments added above here might make this clearer now.
I'd appreciate feedback if any of it doesn't make sense... I'm so close to it now!